sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -351,10 +351,24 @@ object PartitioningUtils {
    */
   def getPathFragment(spec: TablePartitionSpec, partitionSchema: StructType): String = {
```

Comment:
Can we use the same code path with the one when it's non-empty?

Reply:
cc @MaxGekk FYI

Reply:
> Can we use the same code path with the one when it's non-empty?

I think no. The below code separate code path for update partition metadata of updated partition (non-empty) and empty

https://github.com/apache/spark/blob/badb0393d46d7aef90710e51e233fb5077977423/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelationCommand.scala#L189-L198

Reply:
Can you factor out PartitioningUtils.parsePartitions and share the same casting logic?

Reply:
PartitioningUtils.parsePartitions use castPartValueToDesiredType to cast multiple type of partition. I think in this issue, we only need cast number type (or remove leading zeros), the other types will be handled by org.apache.spark.sql.catalyst.expressions.CastBase